datadog_lambda/dsm.py

```diff
@@ -1,38 +1,87 @@
-from datadog_lambda import logger
+import logging
+import json
 from datadog_lambda.trigger import EventTypes
 
+logger = logging.getLogger(__name__)
 
-def set_dsm_context(event, event_source):
 
+def set_dsm_context(event, event_source):
     if event_source.equals(EventTypes.SQS):
         _dsm_set_sqs_context(event)
 
 
 def _dsm_set_sqs_context(event):
-    from datadog_lambda.wrapper import format_err_with_traceback
-    from ddtrace.internal.datastreams import data_streams_processor
-    from ddtrace.internal.datastreams.processor import DsmPathwayCodec
-    from ddtrace.internal.datastreams.botocore import (
-        get_datastreams_context,
-        calculate_sqs_payload_size,
-    )
-
     records = event.get("Records")
     if records is None:
         return
-    processor = data_streams_processor()
 
     for record in records:
         try:
-            queue_arn = record.get("eventSourceARN", "")
+            arn = record.get("eventSourceARN", "")
+            context_json = _get_dsm_context_from_sqs_lambda(record)
+            if not context_json:
+                continue
+            _set_dsm_context_for_record(context_json, "sqs", arn)
+
+        except Exception as e:
+            logger.error(f"Unable to set dsm context: {e}")
+
+
+def _set_dsm_context_for_record(context_json, type, arn):
+    from ddtrace.data_streams import set_consume_checkpoint
```

Comment:
We need to be sure to set a minimum version for the ddtrace dependency. To do that, you'll want to find the first version of ddtrace that includes this `set_consume_checkpoint`. Then update pyproject.toml with this version.